Subject: Remove superfluous (Christoph Biedl)
Origin: upstream, commit FILE5_46-69-g280e121f <https://github.com/file/file/commit/FILE5_46-69-g280e121f>
Author: Christos Zoulas <christos@zoulas.com>
Date: Thu Mar 20 17:45:56 2025 +0000

--- a/magic/Magdir/zip
+++ b/magic/Magdir/zip
@@ -17,7 +17,6 @@
 >>6	leshort		x		\b, extract using at least
 >>6	use		zipversion
 # This is DOS date like: ledate 21:00:48 19 Dec 2001 != DOS 00:00 1 Jan 2010 ~ 0000213C
->>12	ulelong		x		\b, last modified
 >>14	lemsdosdate	x		\b, last modified %s
 >>12	lemsdostime	x		%s
 # uncompressed size of 1st entry; FFffFFff means real value stored in ZIP64 record
